uk.co.compucorp.civicrm.hrcore/CRM/HRCore/Upgrader/Steps/1000.php

```diff
@@ -113,19 +113,15 @@ private function up1000_setAvailableCountries() {
    * Sets Available Provinces to 'all provinces'
    */
   private function up1000_setAvailableProvinces() {
-    // ToDo : Fetch Provinces via API after upgrading to newer civicrm version since it is not available on civicrm 4.7.9
-    $tableName = CRM_Core_DAO_StateProvince::getTableName();
-    $query = CRM_Core_DAO::executeQuery("SELECT id FROM {$tableName}");
+    $countries = civicrm_api3('Country', 'get', ['sequential' => 1]);
```

Comment:
Using province ID's was breaking the MySQL client I believe, because although the field where this is stored is a TEXT field, a fatal exception was being thrown due to the size of the string being stored.  I investigated how the ProvinceLimit setting is supposed to be used, but documentation for the field is somewhat obscure.  Still, the few on-line usages I found of the setting always used country codes, instead of province codes.  So basically, the setting is used to filter out provinces by providing the ID's for the countries for which you want provinces to be used in the system.

This is the available documentation for the setting in the API:
```
{
            "group_name": "Localization Preferences",
            "group": "localization",
            "name": "provinceLimit",
            "type": "Array",
            "quick_form_type": "Element",
            "html_type": "advmultiselect",
            "html_attributes": {
                "size": 5,
                "style": "width:150px",
                "class": "advmultiselect"
            },
            "default": [

            ],
            "add": "4.3",
            "title": "Available States and Provinces (by Country)",
            "is_domain": 1,
            "is_contact": 0,
            "description": "",
            "help_text": null,
            "pseudoconstant": {
                "callback": "CRM_Admin_Form_Setting_Localization::getAvailableCountries"
            }
        },
```